File: apps/spring-boot-api/src/main/java/org/bootstrapbugz/api/shared/error/exception/BadRequestException.java
package org.bootstrapbugz.api.shared.error.exception;

import java.io.Serial;

import org.bootstrapbugz.api.shared.error.ErrorDomain;
import org.springframework.http.HttpStatus;

import lombok.Getter;

@Getter
public class BadRequestException extends RuntimeException {
  @Serial private static final long serialVersionUID = -6237654540916338509L;
  private final ErrorDomain domain;
  private final HttpStatus status;

  public BadRequestException(String message, ErrorDomain domain) {
    super(message);
    this.domain = domain;
    this.status = HttpStatus.BAD_REQUEST;
  }
}

File: backend/spring-boot-kotlin/src/main/java/org/bootstrapbugz/api/shared/error/exception/BadRequestException.kt
package org.bootstrapbugz.api.shared.error.exception

import org.springframework.http.HttpStatus
import java.io.Serial

class BadRequestException(val field: String, message: String?) : RuntimeException(message) {
    val status: HttpStatus = HttpStatus.BAD_REQUEST

    companion object {
        @Serial
        private val serialVersionUID = -6237654540916338509L
    }
}
